src/D2Reader/Models/Item.cs

```diff
@@ -18,8 +19,50 @@ public class Item
         internal bool IsEthereal() => HasFlag(ItemFlag.Ethereal);
         internal bool HasRuneWord() => HasFlag(ItemFlag.Runeword);
 
-        internal bool IsInCube() => ItemData.InvPage == InventoryPage.HoradricCube;
-        internal bool IsInStash() => ItemData.InvPage == InventoryPage.Stash;
-        internal bool IsInInventory() => ItemData.InvPage == InventoryPage.Inventory;
+        internal bool IsInCube() => !IsEquipped() && ItemData.InvPage == InventoryPage.HoradricCube;
+        internal bool IsInStash() => !IsEquipped() && ItemData.InvPage == InventoryPage.Stash;
+        internal bool IsInInventory() => !IsEquipped() && ItemData.InvPage == InventoryPage.Inventory;
+    }
+
+    public class ItemLocation
+    {
+        public int X;
+        public int Y;
+        public int Width;
+        public int Height;
+        public BodyLocation BodyLocation;
+        public ItemContainer Container;
+
+        public static bool operator ==(ItemLocation left, ItemLocation right)
+        {
+            if ((object)left == null)
+                return (object)right == null;
+
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(ItemLocation left, ItemLocation right)
+        {
+            return !(left == right);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null || this.GetType() != obj.GetType())
+                return false;
+
+            var other = (ItemLocation)obj;
+            return this.X            == other.X &&
+                   this.Y            == other.Y &&
+                   this.Width        == other.Width &&
+                   this.Height       == other.Height &&
+                   this.BodyLocation == other.BodyLocation &&
+                   this.Container    == other.Container;
```

Comment:
No need to change, but just a note because i saw it in multiple places:

So far we didn't use this kind of alignment of parameters or operators, and at least the reason for me is that a change in one line may increase PR unecessary, because all lines need to be adjusted if longest line changes. But as said, no need to change, it may be better readable this way, and we read this more often than change it :D

And as this is new code and not purely formatting, this is fine for me!